### PR TITLE
openapi: correct job error message on incorrect apiFile

### PR DESCRIPTION
**Pull request by rm-star**
…ed the extraneous URL argument now the offending filename is displayed in the error output

Fixes zaproxy/zaproxy#7370

Signed-off-by: Wil Hadden <wilhadden@gmail.com>

### CODE REVIEW

**Comment by rm-star** [comment]
Hi,

Hope this isn't an automated email.
I thought I had signed it off, I used the -s option like your docs said and
I can see the text of the sign off in the commit. Do you know if there's
something else I should do?

Thanks
Wil

**Comment by kingthorin** [comment]
I see that the commit message contains the text but it doesn’t seem that the commit is actually signed off. I’m not sure how that pans out with git, I know I’ve encountered it before.

You can just:
`git commit --amend --no-edit --signoff` then force push.

**Comment by kingthorin** [comment]
Thanks!

**Review by kingthorin — APPROVED** [review]

**Comment by kingthorin** [comment]
Could you also add a note to the CHANGELOG.md file (https://github.com/zaproxy/zap-extensions/blob/main/addOns/openapi/CHANGELOG.md). In the `Unreleased` section under `### Fixed`. Something like:

> Corrected automation job error output when importing (Issue 7370).

**Comment by thc202** [comment]
You might want to reset the author of the commit, still using a different email.

**Comment by rm-star** [comment]
I've added a CHANGELOG entry.
Thanks for letting me know about not needing to raise an issue, I wasn't
sure if I needed one or not

**Comment by rm-star** [comment]
I can get on to that tomorrow.

**Comment by kingthorin** [comment]
Thanks for adding the changelog entry. Sorry to be a pain but could you tweak it, there was already a `### Fixed` heading :wink: No rush, tackle it with the unit test or whatever...

**Comment by rm-star** [comment]
Haha yeah no worries.

File: addOns/openapi/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,9 @@ All notable changes to this add-on will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
+### Fixed
+- Error message when apiFile field is not accessible was outputting the targetUrl and not the incorrect filename
```
Comment (psiinon) [inline addOns/openapi/CHANGELOG.md:8]: This line should be added after the existing "Fixed" section :) ie immediately under "JSON body examples..."

File: addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/automation/OpenApiJobUnitTest.java [hunk 2]
```diff
@@ -150,7 +151,7 @@ void shouldFailIfInvalidUrl() {
     @Test
     void shouldFailIfInvalidFile() {
         // Given
-        Constant.messages = new I18N(Locale.ENGLISH);
+        TestUtils.mockMessages(new ExtensionOpenApi());
```
Comment (thc202) [inline addOns/openapi/src/test/java/org/zaproxy/zap/extension/openapi/automation/OpenApiJobUnitTest.java:154]: `mockMessages(new ExtensionOpenApi())`

File: addOns/openapi/CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ## Unreleased
 ### Fixed
 - JSON body examples specified under `schema` were being enclosed in quotes.
+- Error message when apiFile field is not accessible was outputting the targetUrl and not the incorrect filename
```
Comment (thc202) [inline addOns/openapi/CHANGELOG.md:9]:
```suggestion
- Error message when apiFile field is not accessible was outputting the targetUrl and not the incorrect filename (Issue 7370).
```

File: addOns/openapi/CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ## Unreleased
 ### Fixed
 - JSON body examples specified under `schema` were being enclosed in quotes.
+- Error message when apiFile field is not accessible was outputting the targetUrl and not the incorrect filename (Issue 7370).
```
Comment (thc202) [inline addOns/openapi/CHANGELOG.md:9]:
```suggestion
- Error message when `apiFile` field is not accessible was outputting the `targetUrl` and not the incorrect filename (Issue 7370).
```

**Comment by thc202** [comment]
Squashed the commits.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Comment by rm-star** [comment]
Thanks for the merge!

**Comment by thc202** [comment]
How would you like to be credited (e.g. name, handle)?
https://www.zaproxy.org/docs/desktop/credits/#zap-extended-team

**Comment by rm-star** [comment]
Wil Hadden (https://twitter.com/WilHadden)

Thanks!

**Comment by rm-star** [comment]
I have altered an existing test to cover my changes and updated the
CHANGELOG now

**Pull request merged**